packages/compass-data-modeling/src/components/diagram-editor.tsx

```diff
@@ -198,7 +211,30 @@ const DiagramEditor: React.FunctionComponent<{
     });
   }, [model?.relationships]);
 
-  const nodes = useMemo(() => {
+  const applyInitialLayout = useCallback(async () => {
+    try {
+      const { nodes: positionedNodes } = await applyLayout(
+        nodes,
+        edges,
+        'LEFT_RIGHT'
+      );
+      onApplyInitialLayout(
+        positionedNodes.reduce((obj, node) => {
+          obj[node.id] = [node.position.x, node.position.y];
+          return obj;
+        }, {} as Record<string, [number, number]>)
```

Comment:
stylistic nit, feel free to ignore:

```suggestion
        Object.fromEntries(
          positionedNodes.map((node) => [
            node.id,
            [node.position.x, node.position.y],
          ])
        )
```